src/__snapshots__/core.test.ts.snap

```diff
@@ -15,6 +15,10 @@ declare module "nextjs-routes" {
 
   type Query<Params = {}> = Params & { [key: string]: string | string[] | undefined };
 
+  export type RoutedQuery = {
+    [K in Route as K["pathname"]]: Exclude<K["query"], undefined>;
+  };
```

Comment:
```suggestion
  type QueryForPathname = {
    [K in Route as K["pathname"]]: Exclude<K["query"], undefined>;
  };
  
  export type RoutedQuery<Pathname extends Route["pathname"] = Route["pathname"]> = QueryForPathname[Pathname]
```
Could you export a generic type instead? That way the syntax is consistent for end users:

```
RoutedQuery<"/foos/[foo]">
```
vs
```
RoutedQuery["/foos/[foo]"]
```